flanker/mime/message/part.py

```diff
@@ -665,18 +665,32 @@ def stronger_encoding(a, b):
 
 
 def has_long_lines(text, max_line_len=599):
-    '''
+    """
     Returns True if text contains lines longer than a certain length.
     Some SMTP servers (Exchange) refuse to accept messages "wider" than
     certain length.
-    '''
+    """
     if not text:
         return False
     for line in text.splitlines():
         if len(line) >= max_line_len:
             return True
     return False
 
+def _base64_decode(s):
+    """Recover base64 if it is broken."""
+    try:
+        return base64.b64decode(s)
+    except TypeError:
```

Comment:
Checking the length of the string is O(1) so you do not need to wait for exception to fix these types of errors. I suggest:
```python
def _base64_decode(s):
    tail_size = len(s) % 4
    if tail_size == 0:
        return base64.b64decode(s)

    if tail_size == 1:
        # crop last character as adding padding does not help
        return base64.b64decode(s[:-1])
 
    # add padding
    return base64.b64decode(s + "=" * (4 - tail_size))
```